services/data_sources/json_build_data_source.rb

```diff
@@ -22,6 +25,42 @@ def self.json_to_attribute_name_proc_map
       }
       return { :@timestamp => seconds_to_datetime_proc }
     end
+
+    def self.attribute_to_type_map
+      return { :@artifacts => Artifact }
+    end
+  end
+
+  # Mixin the JSONConvertible class for Artifact
+  class Artifact
+    include FastlaneCI::JSONConvertible
+
+    def self.json_to_attribute_name_proc_map
+      provider_object_to_provider = proc { |object|
+        nil if object.nil?
```

Comment:
What does that line do? It's not 100% clear

Reply:
So, there might be some chances (but it shouldn't ever happen) that the `Artifact`'s `ArtifactProvider` is `null` in the JSON, so it must be `nil` in the object. But that means that the `Artifact` is unrecoverable.